solver.rb

```diff
@@ -0,0 +1,27 @@
+class Solver
+  def factorial(num)
+    if num.negative?
+      raise ArgumentError, 'negative number not allowed'
+    elsif num.zero?
+      1
+    else
+      num * factorial(num - 1)
+    end
+  end
+
+  def reverse(word)
+    word.reverse
+  end
+
+  def fizzbuzz(number)
+    if (number % 3).zero? && (number % 5).zero?
+      'fizzbuzz'
+    elsif (number % 3).zero?
+      'fizz'
+    elsif (number % 5).zero?
+      'buzz'
+    else
+      number.to_s
+    end
+  end
+end
```

Comment:
- [ ] This is a TDD project, so the commit history sequence should clearly show that each of the Solver class methods was tested before writing the code, kindly make sure to write your tests before implementing the Code

Reply:
Hi @Ismailco,
Thank you for giving us time to share your beneficial suggestion, We will consider it next time.